templates/CRM/Event/Form/Registration/DisplayProfile.tpl

```diff
@@ -14,11 +14,13 @@
             {if $primaryParticipantProfile.CustomPre}
                <fieldset class="label-left no-border"><div class="bold crm-profile-view-title">{$primaryParticipantProfile.CustomPreGroupTitle}</div>
                    {foreach from=$primaryParticipantProfile.CustomPre item=value key=field}
-                      <div class="crm-public-form-item crm-section {$field}-section">
-                          <div class="label">{$field}</div>
-                          <div class="content">{$value}</div>
-                          <div class="clear"></div>
-                      </div>
+                      {if !empty($value)}
```

Comment:
The number `0` is theoretically a valid value for a custom field. I think it should only check for the empty string, i.e. `=== ''`. (Sidenote: at one time we were also avoiding empty() in tpls because it didn't work with smarty secure mode, but the v2 version of that was abandoned so I think we're ok to use empty again now.)

On a larger note, I don't know if this is a case of "one person's bug is another person's feature", meaning I could see some people wanting it to display with an empty value, as visual confirmation that they didn't make a selection, but I've never personally had anyone ask for that. I have seen people ask to have blank custom values removed from emails though. So I'd vote a soft +1 on the overall concept.

Reply:
Good thinking, thanks for that. I've made that amend to check for the empty string.

Reply:
It works, but one tricky part is that now if you don't enter ANY values in the profile section, it still shows the title, which looks a little weird.

Sorry you probably thought this would be an easy one!

One possibility is have two passes over the data, one where it just checks if _anything_ has a value and sets a variable, and then the second pass as you have it now but the entire group block wrapped in an `{if}` that checks your variable. It feels awkward though - maybe there's a better way.

Reply:
Ha, I should never make the mistake of thinking it'll be easy! 

As you suggested, I've tried updating the checks with an additional check that sets a variable. It is a little awkward, but does seem to work.